pkg/provisioner/ironic/testserver/server.go

```diff
@@ -81,6 +82,17 @@ func (m *MockServer) Response(pattern string, payload string) *MockServer {
 	return m
 }
 
+// ResponseJSON marshals the JSON object as payload returned by the response
+// handler
+func (m *MockServer) ResponseJSON(pattern string, payload interface{}) *MockServer {
```

Comment:
You were way less lazy here than I was. We should land your change before #647 and then I will rework that one to use this JSON handling that you've added.